app/user.go

```diff
@@ -1,36 +1,78 @@
 package app
 
 import (
+	"encoding/json"
 	"errors"
+	"os"
 
-	"github.com/ubclaunchpad/cumulus/consensus"
+	crand "crypto/rand"
 
 	"github.com/ubclaunchpad/cumulus/blockchain"
+	"github.com/ubclaunchpad/cumulus/consensus"
 	"github.com/ubclaunchpad/cumulus/msg"
-
-	crand "crypto/rand"
 )
 
 // User holds basic user information.
 type User struct {
-	blockchain.Wallet
+	Wallet    *blockchain.Wallet
 	Name      string
 	BlockSize uint32
 }
 
 // NewUser creates a new user
 func NewUser() *User {
 	return &User{
-		Wallet:    *blockchain.NewWallet(),
+		Wallet:    blockchain.NewWallet(),
 		BlockSize: blockchain.DefaultBlockSize,
+		Name:      "Default User",
+	}
+}
+
+// Public returns the public key (address) of the given user
```

Comment:
Public key and address are not the same. Address is hash of key, is that right @chadlagore ?

Reply:
Good call, I'll remove that "(address)" comment, it's not really necessary.